src/agents/der_aggregator.jl

```diff
@@ -0,0 +1,636 @@
+abstract type AbstractDERA <: Agent end
+
+struct DERAOptions <: AgentOptions
+    solvers::HEMSolver
+    # solvers::Union{HEMSolver, Dict{String, <:HEMSolver}}
+end
+
+"""
+Construct DERAOptions with an MOI.OptimizerWithAttributes instance.
+"""
+function DERAOptions(attributes::MOI.OptimizerWithAttributes)
+    return DERAOptions(AnySolver(attributes))
+end
+
+mutable struct DERA <: AbstractDERA
+    id::String
+    # incentive function for DER aggregation (piece-wise linear function: incentive (x) vs participation (y))
+    dera_stor_incentive_function::DataFrame
+    # parameters applied to aggregated capacities of DER to represent potential friction of aggregation
+    aggregation_friction::ParamArray
+    # incentive level by year
+    incentive_level::ParamArray
+    # aggregation level by year
+    aggregation_level::ParamArray
```

Comment:
Are these variables or parameters?

Reply:
They're variables in the optimization problem. But here, we just define a container that will hold the results of the variables.

Reply:
I highlighted these because I think they might be against current naming convention. However, I wouldn't say we've articulated a clear naming convention, so I'll let this go for now.